src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -2746,15 +2757,22 @@ impl Coordinator {
                 _ => None,
             };
 
+            if matches!(explainee, Explainee::Query) {
+                // We have the opportunity to name an `until` frontier that will prevent work we needn't perform.
+                // By default, `until` will be `Antichain::new()`, which prevents no updates and is safe.
+                if let Some(as_of) = dataflow.as_of.as_ref() {
+                    if !as_of.is_empty() {
+                        if let Some(next) = as_of.as_option().and_then(|as_of| as_of.checked_add(1))
+                        {
+                            dataflow.until = timely::progress::Antichain::from_elem(next);
+                        }
+                    }
+                }
+            }
```

Comment:
This part might be interesting to @vmarcos: It will make it so that the EXPLAIN output will contain the monotonic operators when the "monotonic one-shot SELECTs" feature flag is set.

Reply:
I'm now running `sqllogictest` to see if this changes any plans at the moment.